### PR TITLE
DeviceTests: Added Traits to headless runner

### PR DESCRIPTION
**Pull request by Redth**
### Description of Change ###

Adds the ability to filter device tests by traits.

For starters, the `DeviceType` trait is now excluded based on if the test runner is on a physical or virtual device (so if we're running on `Virtual`, all tests marked with a `DeviceType` trait of `Device` will be skipped (and vice versa).

We can add more traits as needed.

### PR Checklist ###

- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Updated documentation ([see walkthrough](https://github.com/xamarin/Caboodle/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

File: DeviceTests/Caboodle.DeviceTests.Shared/Traits.cs [hunk 1]
```diff
@@ -8,6 +11,9 @@ internal static class DeviceTypes
         {
             public const string Physical = "Physical";
             public const string Virtual = "Virtual";
+
+            internal static string ToExclude =>
+                DeviceInfo.DeviceType == RuntimeDeviceType.Physical ? Virtual : Physical;
```
Comment (mattleibow) [inline DeviceTests/Caboodle.DeviceTests.Shared/Traits.cs]: Something about this wording/idea is a bit annoying to my mind, but I can't really think of a better/different way right now.

Comment (Redth) [reply DeviceTests/Caboodle.DeviceTests.Shared/Traits.cs]: Yeah :( I agree...

File: DeviceTests/Caboodle.DeviceTests.Android/MainActivity.cs [hunk 2]
```diff
@@ -18,7 +21,13 @@ protected override void OnCreate(Bundle bundle)
             var hostPort = Intent.Extras?.GetInt("HOST_PORT", 10578) ?? 10578;
 
             if (!string.IsNullOrEmpty(hostIp))
-                UnitTests.HeadlessRunner.Tests.RunAsync(hostIp, hostPort, typeof(Battery_Tests).Assembly);
+            {
+                var filters = new List<XUnitFilter>
+                {
+                    new XUnitFilter(Traits.DeviceType, Traits.DeviceTypes.ToExclude, true)
```
Comment (mattleibow) [inline DeviceTests/Caboodle.DeviceTests.Android/MainActivity.cs]: We may need to address this in future if we ever have to add more filters - maybe find a way to sync them up or something. We just have to remember to do this for each platform project.

Comment (Redth) [reply DeviceTests/Caboodle.DeviceTests.Android/MainActivity.cs]: Yeah I suppose we could have a shared helper method to `GetSharedFilters()`

**Comment by mattleibow** [comment]
@Redth The style Cop got you! ;)

**Pull request merged**